x/commerciodocs/keeper.go

```diff
@@ -127,22 +127,23 @@ func (keeper Keeper) ShareDocument(ctx sdk.Context, reference string, sender typ
 	sharingStore := ctx.KVStore(keeper.sharingStoreKey)
 
 	// Save the shared document for the sender
-	if !keeper.CanReadDocument(ctx, sender, reference) {
+	if keeper.CanReadDocument(ctx, sender, reference) {
 		sharingStore.Set([]byte(sender), keeper.cdc.MustMarshalBinaryBare(sharing))
+	} else {
+		return sdk.ErrUnauthorized(fmt.Sprintf("The sender with address %s doesnt have the rights on this document", sender))
 	}
 
 	// Save the shared document for the recipient
 	if !keeper.CanReadDocument(ctx, recipient, reference) {
```

Comment:
I think that this code will never be reached. If the condition `keeper.CanReadDocument(ctx, sender, reference)` is `false`, than the `else` branch will be followed and an `sdk.ErrUnauthorized` will be returned, stopping the flow of the method itself. 

Everything below that `if-else` should be deleted IMO.

Reply:
Nope, it shouldn't be deleted cause, if the document belongs to the sender, then we should share that with the recipient (if he hasnt already had it).